### PR TITLE
Fix IsImplicit for anonymous type field assignments

### LINKED ISSUE #22588: AnonymousTypeCreation initializer IsImplicit is not calculated correctly.
Currently, we calculate IsImplicit from the operand of the AnonymousTypeCreationInitializer. However, this isn't correct as the operand can be an implicit conversion, but the assignment isn't. https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb#L200 is the affected line, see https://github.com/dotnet/roslyn/pull/22056/files/fe06e85039aee67749b13c161eeb39f79586bf96#r143312625.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #22588
vso :https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/517766

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs [hunk 1]
```diff
@@ -193,7 +193,7 @@ private ImmutableArray<IOperation> GetAnonymousObjectCreationInitializers(BoundA
                 SyntaxNode syntax = value.Syntax?.Parent ?? expression.Syntax;
                 ITypeSymbol type = target.Type;
                 Optional<object> constantValue = value.ConstantValue;
-                var assignment = new SimpleAssignmentExpression(target, value, _semanticModel, syntax, type, constantValue, isImplicit: value.IsImplicit);
+                var assignment = new SimpleAssignmentExpression(target, value, _semanticModel, syntax, type, constantValue, isImplicit: false);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs]: >false [](start = 136, length = 5)

expression.WasCompilerGenerated?

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb [hunk 2]
```diff
@@ -207,7 +207,7 @@ Namespace Microsoft.CodeAnalysis.Operations
                 Dim syntax As SyntaxNode = If(value.Syntax?.Parent, expression.Syntax)
                 Dim type As ITypeSymbol = target.Type
                 Dim constantValue As [Optional](Of Object) = value.ConstantValue
-                Dim assignment = New SimpleAssignmentExpression(target, value, _semanticModel, syntax, type, constantValue, isImplicit:=value.IsImplicit)
+                Dim assignment = New SimpleAssignmentExpression(target, value, _semanticModel, syntax, type, constantValue, isImplicit:=False)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb]: >False [](start = 136, length = 5)

expression.WasCompilerGenerated?

**Comment by AlekseyTs** [comment]
Please add unit-tests for C# that would make the behavior change observable.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1).

**Review by 333fred — APPROVED** [review]

**Review by AlekseyTs — APPROVED** [review]
LGTM

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler Can I get one more review please?

File: src/Compilers/CSharp/Test/Symbol/Symbols/AnonymousTypesSemanticsTests.cs [hunk 3]
```diff
@@ -1610,6 +1610,65 @@ public static void Main()
 }", additionalRefs: new[] { SystemCoreRef }).VerifyDiagnostics();
         }
 
+        [CompilerTrait(CompilerFeature.IOperation)]
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/AnonymousTypesSemanticsTests.cs]: Consider adding WorkItem

**Review by jcouv — APPROVED** [review]
LGTM

**Comment by jinujoseph** [comment]
pre-approved to merge via [link](https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/517766)

**Pull request merged**